podcast/2.18.md

```diff
@@ -0,0 +1,545 @@
+
+**Beyang Liu:**:
+
+Yeah. It pops up everywhere. Just real quick, you mentioned your affinity for Monorepos. I have to say I myself am in the Monorepo camp. Can you talk a bit about why you prefer that model for tracking source code as compared to the multi-repo, more GitHub-driven model?
+
+**Kelly Norton:**
+
+Yeah. I think this really becomes acute the moment you introduce client libraries or protobuffs into your ... So anytime you start defining the interfaces between things, boy, does it become acute really quickly. So I always have a joke that anytime the question of, "Hey, we're going to start doing services and we use a GitHub model." And I go, "But where are you going to store your protobuff definitions?" And inevitably, everyone laughs because there has been a long, drawn-out argument about where the protobuff definitions live. And I mean, my basic pitch for a Monorepo is it's actually really hard to anticipate where the complexities are going to come in the future in terms of where code lives and what it takes to change something in one place and get it to the other place.
+
+Monorepo is like, I mean, they basically solved that problem because everything's in the same place. And I think people ... I mean the argument I had heard against them after leaving Google was that, "Well the Google model really requires a lot of tooling." And originally, I was sympathetic to that. And I go, "Yeah, it did require a lot of tooling." And then after doing the non-Google model for a little while, I realized like, "Wow, we have a lot of tooling to support this sort of multi-repo world." So would really just rather have the simpler mental model about how to make code changes and just invest in the tooling because I invested in the tooling anyway; we still built a ton of tooling.
+
+To give you a concrete example inside of Mailchimp: we have some PHP libraries and we distribute them via Composer internally. So, sadly, even though that fortunately Composer exists – and Composer is kind of like the package manager for PHP, and it's good; it does its job well – it's really intended for pulling in open-source projects, but it does the job. But it just adds so much additional complexity to our changes to our shared libraries because we have to make a change to the library. And then we actually have automations that then push pull requests to all the various apps that use that library.
```

Comment:
```suggestion
To give you a concrete example inside of Mailchimp: we have some PHP libraries and we distribute them via Composer internally. So, sadly, even though fortunately Composer exists – and Composer is kind of like the package manager for PHP, and it's good; it does its job well – it's really intended for pulling in open-source projects, but it does the job. But it just adds so much additional complexity to our changes to our shared libraries because we have to make a change to the library. And then we actually have automations that then push pull requests to all the various apps that use that library.
```